pennylane/ops/functions/equal.py

```diff
@@ -336,112 +354,157 @@ def _equal_operators(
         return True
 
     if op1.arithmetic_depth != op2.arithmetic_depth:
-        return False
+        return f"op1 and op2 have different arithmetic depths. Got {op1.arithmetic_depth} and {op2.arithmetic_depth}"
 
     if op1.arithmetic_depth > 0:
         # Other dispatches cover cases of operations with arithmetic depth > 0.
         # If any new operations are added with arithmetic depth > 0, a new dispatch
         # should be created for them.
-        return False
-    if not all(
-        qml.math.allclose(d1, d2, rtol=rtol, atol=atol) for d1, d2 in zip(op1.data, op2.data)
-    ):
-        return False
+        return f"op1 and op2 have arithmetic depth > 0. Got arithmetic depth {op1.arithmetic_depth}"
+
     if op1.wires != op2.wires:
-        return False
+        return f"op1 and op2 have different wires. Got {op1.wires} and {op2.wires}."
 
     if op1.hyperparameters != op2.hyperparameters:
-        return False
+        return (
+            "The hyperparameter is not equal for op1 and op2.\n"
+            f"Got {op1.hyperparameters}\n and {op2.hyperparameters}."
+        )
+
+    if not all(
+        qml.math.allclose(d1, d2, rtol=rtol, atol=atol) for d1, d2 in zip(op1.data, op2.data)
+    ):
+        return f"op1 and op2 have different data.\n" f"Got {op1.data} and {op2.data}"
```

Comment:
Surprised this didn't cause a pylint error. Optional suggestion below:

```suggestion
        return f"op1 and op2 have different data.\nGot {op1.data} and {op2.data}"
```